File: 진우/26weeks/1547. 공/공.cc
#include <iostream>

using namespace std;

int main() {
    int m, a, b;
    cin >> m;  // 컵을 섞는 횟수

    int num = 1; // 현재 공이 있는 컵 번호

    for (int i = 0; i < m; i++) {
        cin >> a >> b;  // 바꿀 컵 번호

        // 공이 a에 있으면 b로 이동
        if (num == a) {
            num = b;
        }

        // 공이 b에 있으면 a로 이동
        else if (num == b) {
            num = a;
        }
    }

    cout << num; // 공이 들어있는 컵 번호 출력

    return 0;
}

File: 진우/26weeks/1547. 공/공.py
# 컵을 섞는 횟수 입력
m = int(input())

# 현재 공이 들어있는 컵 번호 (처음에는 1번 컵)
num = 1

# m번 동안 컵 위치를 바꾸는 작업 수행
for _ in range(m):
    # 서로 위치를 바꿀 두 컵 번호 입력
    a, b = map(int, input().split())

    # 공이 a번 컵에 있었다면 b번 컵으로 이동
    if num == a:
        num = b

    # 공이 b번 컵에 있었다면 a번 컵으로 이동
    elif num == b:
        num = a

# 모든 섞기가 끝난 후 공이 들어있는 컵 번호 출력
print(num)